### DISCUSSION TITLE
Bind to root of project

### DISCUSSION THREAD

**Discussion by kapral18** [nvim-neo-tree/neo-tree.nvim#1378] [Q&A]
I want to make neo-tree always open the root of the project instead of cwd.

Sometimes, when I am using https://github.com/folke/persistence.nvim, and restore the session neotree changes the root to the folder of restored file, which is suboptimal. I want to always bind to the root. I want to be able to calculate it like any file with .git or yarn.lock or package-lock etc... How can I achieve that with neo-tree?

I checked the dir option, but the dir option is supposed to be a vim.fn.expand string which only allows to use file modifiers. And I want something that will allow me to use vim.fs.find('.git', {upward: true, type: "directory"...}) kind of stuff.

Thanks

**Comment by pysan3** [comment]
Neotree should restore the cwd as-is, so I wonder why persistence.nvim does not save the root dir.

If you want to restore with the project root, I advise you to try to make persistence _save_ the root dir in the first place and not try to do something on _restore_.

We are having discussions on how to restore neo-tree's state when restored with [auto-session](https://github.com/rmagatti/auto-session), so this might give you a hint.

**Reply by kapral18** [reply]
im afraid it's not only session based, for example now if I open 

elastic/kibana in the root explorer shows me the root
and let's say I switch to this file `https://github.com/elastic/kibana/blob/main/x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/data_quality_summary/errors_viewer/helpers.tsx`

first the neo-tree explorer follows the file and shows it with cwd staying at the root of project

but then if I toggle close and open again the explorer it now shows the root starting at 

`impl/`

my config for ref: https://github.com/kapral18/dotfiles/blob/aecb4a64df2cbc2f4550d5083bdd51738f7f36a4/home/dot_config/exact_nvim/exact_lua/exact_plugins/neo-tree.lua

**Reply by pysan3** [reply]
If that's the case, please raise an issue with reproducible config (not a discussion). 

And ping me there.
